velox/functions/sparksql/String.cpp

```diff
@@ -144,6 +424,49 @@ std::shared_ptr<exec::VectorFunction> makeLength(
   return kLengthFunction;
 }
 
+std::vector<std::shared_ptr<exec::FunctionSignature>> concatWsSignatures() {
+  return {// The second and folowing arguments are varchar or array(varchar).
+          // Use "any" to allow the mixed using of these two types. The
+          // argument type will be checked in makeConcatWs.
+          //
+          // varchar, [varchar], [array(varchar)], ... -> varchar.
+          exec::FunctionSignatureBuilder()
+              .returnType("varchar")
+              .constantArgumentType("varchar")
```

Comment:
This signature says that separator must be constant, but it seems that implementation also allows non-constant separator.